src/Lykke.Job.TransactionHandler/Handlers/EthereumCoreCommandHandler.cs

```diff
@@ -0,0 +1,468 @@
+                    case CoinEventType.TransferCompleted:
+                    case CoinEventType.CashoutCompleted:
+                        await ProcessOutcomeOperation(command);
+                        break;
+                    case CoinEventType.CashoutFailed:
+                        await ProcessFailedCashout(command, eventPublisher);
+                        break;
+                    default:
+                        break; ;
+                }
+
+                return CommandHandlingResult.Ok();
+            }
+            catch (Exception e)
+            {
+                _log.WriteError(command.TransactionHash ?? "Empty", command, e);
+                throw;
+            }
+}
+
+        public async Task<CommandHandlingResult> Handle(EnrollEthCashinToMatchingEngineCommand command, IEventPublisher eventPublisher)
+        {
+            try
+            {
+                var cashinId = command.CashinOperationId.ToString("N");
+                var clientId = command.ClientId;
+                var hash = command.TransactionHash;
+                var amount = command.Amount;
+                var asset = await _assetsServiceWithCache.TryGetAssetAsync(command.AssetId);
+                var createPendingActions = command.CreatePendingActions;
+                var clientAddress = command.ClientAddress;
+                var paymentTransaction = PaymentTransaction.Create(hash,
+                    CashInPaymentSystem.Ethereum, clientId, (double)amount,
+                    asset.DisplayId ?? asset.Id, status: PaymentStatus.Processing);
+
+                var exists = await _paymentTransactionsRepository.CheckExistsAsync(paymentTransaction);
+
+                if (exists)
+                {
+                    _log.WriteWarning(command.TransactionHash ?? "Empty", command,
+                            $"Transaction already handled {hash}");
+
+                    return CommandHandlingResult.Ok();
+                }
+
+                if (createPendingActions)
+                {
+                    if (asset.IsTrusted)
+                    {
+                        await _ethererumPendingActionsRepository.CreateAsync(clientId, Guid.NewGuid().ToString());
+                    }
+                }
+
+                var result = await _matchingEngineClient.CashInOutAsync(cashinId, clientId, asset.Id, (double)amount);
+
+                if (result == null ||
+                   (result.Status != MeStatusCodes.Ok &&
+                    result.Status != MeStatusCodes.Duplicate))
+                {
+                    _log.WriteWarning(command.TransactionHash ?? "Empty", result, "ME error");
+
+                    return CommandHandlingResult.Fail(TimeSpan.FromMinutes(1));
+                }
+                else
+                {
+                    eventPublisher.PublishEvent(new EthCashinEnrolledToMatchingEngineEvent()
+                    {
+                        CashinOperationId = command.CashinOperationId,
```

Comment:
Why you send CashinOperationId in this command? This data is already in your aggregate and this is not ID of the aggregate.

Reply:
fixed